From dfe4dada71b2eb177c7e11b801f714be378d98ae Mon Sep 17 00:00:00 2001
From: rewine <luhongxu@deepin.org>
Date: Wed, 24 Jul 2024 16:25:35 +0800
Subject: [PATCH 5/5] Relax wayland-server to 1.22

---
 meson.build | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/meson.build b/meson.build
index e349f3dd..9885e9d8 100644
--- a/meson.build
+++ b/meson.build
@@ -105,7 +105,7 @@ internal_config = configuration_data()
 
 wayland_project_options = ['tests=false', 'documentation=false']
 wayland_server = dependency('wayland-server',
-	version: '>=1.23',
+	version: '>=1.22',
 	fallback: 'wayland',
 	default_options: wayland_project_options,
 )
-- 
2.44.0

