From 293b0aa502d2e64d68290b1b615e9bd6d203d818 Mon Sep 17 00:00:00 2001
From: groveer <groveer@outlook.com>
Date: Mon, 6 May 2024 15:42:18 +0800
Subject: [PATCH] wlr_pointer_gestures_v1: emit destroy signal in destroy
 function

---
 types/wlr_pointer_gestures_v1.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/types/wlr_pointer_gestures_v1.c b/types/wlr_pointer_gestures_v1.c
index bd9e1f9eb..ebb1c8689 100644
--- a/types/wlr_pointer_gestures_v1.c
+++ b/types/wlr_pointer_gestures_v1.c
@@ -396,6 +396,7 @@ static void pointer_gestures_v1_bind(struct wl_client *wl_client, void *data,
 static void handle_display_destroy(struct wl_listener *listener, void *data) {
 	struct wlr_pointer_gestures_v1 *gestures =
 		wl_container_of(listener, gestures, display_destroy);
+	wl_signal_emit_mutable(&gestures->events.destroy, NULL);
 	wl_list_remove(&gestures->display_destroy.link);
 	wl_global_destroy(gestures->global);
 	free(gestures);
--
GitLab
