diff --git a/types/output/cursor.c b/types/output/cursor.c
index ad37a04..341692d 100644
--- a/types/output/cursor.c
+++ b/types/output/cursor.c
@@ -359,6 +359,8 @@ static bool output_cursor_attempt_hardware(struct wlr_output_cursor *cursor) {
 		return false;
 	}
 
+	output->hardware_cursor = NULL;
+
 	struct wlr_texture *texture = cursor->texture;
 
 	// If the cursor was hidden or was a software cursor, the hardware
